src/react-components/input/Slider.js

```diff
@@ -0,0 +1,143 @@
+import React from "react";
+import { useState, useCallback, useEffect, useRef } from "react";
+import PropTypes from "prop-types";
+import classNames from "classnames";
+import styles from "./Slider.scss";
+import { trackColor } from "./Slider.scss";
+
+function round(step, n) {
+  return Math.round(n / step) * step;
+}
+
+function getLinearGradientCSS(ratio, leftColor, rightColor) {
+  return [
+    "-webkit-gradient(",
+    "linear, ",
+    "left top, ",
+    "right top, ",
+    "color-stop(" + ratio + ", " + leftColor + "), ",
+    "color-stop(" + ratio + ", " + rightColor + ")",
+    ")"
+  ].join("");
+}
+
+function updateGradient(el, min, max, value) {
+  if (el) {
+    const ratio = ((value - min) / (max - min)).toFixed(2);
+    el.style.backgroundImage = getLinearGradientCSS(ratio, trackColor, "transparent");
+  }
+}
+
+export function Slider({ min, max, step, digits, value, defaultValue, onChange, disabled, className }) {
+  const myRoot = useRef();
+  const inputRef = useRef();
+  const [displayValue, setDisplayValue] = useState(-1);
+  const [isDragging, setIsDragging] = useState(false);
+
+  let currentValue = defaultValue;
+  if (isDragging) {
+    currentValue = displayValue !== -1 ? displayValue : defaultValue;
+  } else {
+    currentValue = value !== undefined ? value : displayValue !== -1 ? displayValue : defaultValue;
```

Comment:
This feels extra defensive. If the proptype is a number and you don't provide it a number that's a bug in the calling component. But also as stated above I think defaultValue is kind of an odd concept which is also part of what makes this line feel odd.

Reply:
Same as above.